File: 해시/1845_폰켓몬.cpp
// https://school.programmers.co.kr/learn/courses/30/lessons/1845

#include <vector>
#include <unordered_set>

using namespace std;

int solution(vector<int> nums)
{
    int answer = nums.size() / 2;
    unordered_set<int> unSet;

    for(auto num : nums) {
        unSet.insert(num);
    }

    if(answer > unSet.size()) {
        answer = unSet.size();
    }

    return answer;
}

File: 해시/1845_폰켓몬.java
// https://school.programmers.co.kr/learn/courses/30/lessons/1845

import java.util.*;
import java.io.*;

class Solution {
    public int solution(int[] nums) {
        int answer = nums.length / 2;
        HashSet<Integer> hashSet = new HashSet<>();

        for (int num : nums) {
            hashSet.add(num);
        }

        if(hashSet.size() > answer)
            return answer;
        else
            return hashSet.size();
    }
}